test/distributed/test_functional_api.py

```diff
@@ -458,6 +458,16 @@ def test_all_gather_into_tensor_coalesced(self):
         self.assertEqual(torch.ones([4 * dist.get_world_size()]), res[0])
         self.assertEqual(torch.ones([4 * dist.get_world_size()]) + 1, res[1])
 
+    @skip_if_lt_x_gpu(WORLD_SIZE)
+    @requires_nccl()
+    @with_comms()
+    def test_tracing(self):
```

Comment:
I am a bit surprised this issue was not captured in CI, we have bunch of tests should guard it https://github.com/pytorch/pytorch/blob/main/test/distributed/test_inductor_collectives.py#L572

but thanks for adding more tests to guard it!